data/types.go

```diff
@@ -208,11 +212,79 @@ func (f TargetFileMeta) HashAlgorithms() []string {
 }
 
 type Targets struct {
-	Type        string      `json:"_type"`
-	SpecVersion string      `json:"spec_version"`
-	Version     int         `json:"version"`
-	Expires     time.Time   `json:"expires"`
-	Targets     TargetFiles `json:"targets"`
+	Type        string       `json:"_type"`
+	SpecVersion string       `json:"spec_version"`
+	Version     int          `json:"version"`
+	Expires     time.Time    `json:"expires"`
+	Targets     TargetFiles  `json:"targets"`
+	Delegations *Delegations `json:"delegations,omitempty"`
+}
+
+type Delegations struct {
+	Keys  map[string]*Key `json:"keys"`
+	Roles []DelegatedRole `json:"roles"`
+}
+
+// DelegatedRole enforces the spec 1.0.19 section 4.5 'role MUST specify only one of the "path_hash_prefixes" or "paths"'
+// by forcing one of path_hash_prefixes, paths  with MatchWithHashPrefixes and PathPatterns fields.
+// delegatedRoleJSON and delegatedRoleCopy help decoding and encoding jsons.
+// DelegatedRole UnmarshalJSON will fail and return ErrPathsAndPathHashesSet if both fields are set and not empty
+type DelegatedRole struct {
+	Name             string   `json:"name"`
+	KeyIDs           []string `json:"keyids"`
+	Threshold        int      `json:"threshold"`
```

Comment:
Can we make this a `uint`?

Reply:
All existing Threshold are ints, so for consistency I prefer keeping it as a int